src/general/StringUtils.cs

```diff
@@ -41,4 +43,51 @@ public static string FormatNumber(this double number, bool withSuffix = true)
 
         return number.ToString("0.#", CultureInfo.CurrentCulture);
     }
+
+    /// <summary>
+    ///   Splits string into different chunks by whitespace.
+    /// </summary>
+    /// <param name="input">String to split.</param>
+    /// <param name="ignoreWithinQuotes">Ignore whitespaces within quotation marks.</param>
+    public static string[] SplitByWhiteSpace(string input, bool ignoreWithinQuotes)
+    {
+        var result = new List<string>();
+
+        var cutPoint = 0;
+        var insideQuote = false;
+
+        for (var i = 0; i < input.Length; ++i)
+        {
+            var character = input[i];
+
+            var validQuotes = character != '\\' && i + 1 < input.Length && input[i + 1] == '"';
+
+            if (validQuotes && !insideQuote && ignoreWithinQuotes)
+            {
+                insideQuote = true;
+            }
+            else if (validQuotes && insideQuote)
+            {
+                insideQuote = false;
+            }
+
+            if (character == ' ' && !insideQuote)
+            {
+                if ((i == 0) || (i + 1 < input.Length && input[i + 1] == ' '))
+                {
+                    cutPoint++;
+                    continue;
+                }
+
+                result.Add(Regex.Unescape(input.Substr(cutPoint, i - cutPoint)));
```

Comment:
To me it looks like this is unusable method here https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.unescape?view=net-5.0 it seems to do much else than is required.
Basically a custom helper method that *only* does basic index for the substring changing if `\` is found here.
Also the substr call should have the adjusted cut location for escaping to save on one string instance creation.

Reply:
In my opinion the simplest alternative could be calling string `Replace` after `input.Substr` for replacing occurrences of `\"` with `"`, this makes sense because `SplitByWhiteSpace` method only handle the custom escaped double quotes and nothing else.

Reply:
That wouldn't handle `"text here \\"` that would be parsed incorrectly. This needs a proper parser with state tracking if next character should be ignored.

Reply:
That would require multiple backslashes (more than two) in the source string to have the double backslashes on there (which is not usual in normal use-case). `StringUtils.SplitByWhiteSpace("some text \\", true).ForEach(e => GD.Print(e));` would output `some`, `text` and `\`. The escape sequence would have been handled by the compiler/gettext first before it is given to the method, the use of double quotes escape sequence in the method is the only special case where it requires a multi-backslash to properly have `\"` string constructed.

Reply:
If someone literally writes `\\` on the translations website. That will get escaped so that when the game loads the text it will contain `\\` meaning that our parsing needs to handle that correctly.
In the previous engine I had a very complicated custom written string parsing library with a bunch of tests for edge cases like this. It had things like get the next string in quotes and it would handle nesting of ' and " quotes with escape character handling.

Reply:
I found there's `CUnescape` extension method provided by Godot for strings, looking at its source code it seems to be a couple of replace calls which I think shouldn't be very expensive, and it also handles common escape sequences. I tested with the double backslash and a new line escape sequence and they're unescaped correctly in the output string. I think we could use this?

Reply:
Well maybe. If it is literally just a couple of replaces, we could write our own (there's cost of marshaling data each time C++ code is called from C#, so it's likely faster to just go with a pure C# solution).

Reply:
Added.